diffkemp/simpll/simpll_build.py

```diff
@@ -37,15 +47,16 @@ def get_c_declarations(header_filename):
 llvm_libs = list(filter(lambda x: x != "",
                         llvm_libs.decode("ascii").strip().split()))
 
-simpll_link_arg = "-L{}/diffkemp/simpll".format(get_simpll_build_dir())
+simpll_link_arg = f"-L{get_simpll_build_dir()}/diffkemp/simpll"
```

Comment:
When building DiffKemp locally (running `ninja`, `pip install .`) there is a problem with the `diffkemp` import, maybe we could here use instead of `get_simpll_build_dir()` `os.environ.get("SIMPLL_BUILD_DIR")`  which you provide in the `CMakeFile`, and in the `setup.py` we could set the `SIMPLL_BUILD_DIR` to 'default' `build` directory. This way we would get rid of the `diffkemp` import.

Reply:
At the start of this PR I have been simply copying this function from `diffkemp.util` and it has been working flawlessly.

I have totally forgotten to test the basic `ninja` and `pip` workflow, good catch. Perhaps we could add this workflow to CI? It is possible to setup `virtualenv` and then use `pip` under `nix`.

I am discussing the use of enviromental variable later.

Reply:
Maybe I'm missing something (I just started reviewing the PR) but I was under the impression that we want the standard `cmake`+`ninja` workflow to create the executable and therefore we don't need `pip install .` in the development build.

In other words, I'd prefer having two (supported) ways of building:
1. Development build - just run `cmake`+`ninja` which will generate `build/bin/diffkemp` which can be executed.
2. Release build - run also `pip install` which will create the DiffKemp binary and install it to PATH. We can use this to distribute DiffKemp both via pip and RPM.

Having both of these in the CI makes sense.

Reply:
You got it right, We are generating a new executable in build directory (the 1. Option). We are just discussing that the 2. option (ninja + pip) is currently faulty.

Reply:
> AFAIK, we're not running the `diffkemp` executable anywhere so even if we did run `pip install .` in the release build job, I'm not sure that we'd test everything that should be tested. So, we'll probably need to extend the tests by running the executable.

Maybe for the release build, we could add to the CI building and comparing the `make`-based project located in the `tests/testing_projects/make_based` subdirectory to directly test the `diffkemp` executable.

Reply:
> Maybe for the release build, we could add to the CI building and comparing the `make`-based project located in the `tests/testing_projects/make_based` subdirectory to directly test the `diffkemp` executable.

Yeah, good idea. That's what I'd call an end-to-end test :-)

FWIW, the build part would be similar to what `tests/unit_tests/build_test.py` does, but using CLI.